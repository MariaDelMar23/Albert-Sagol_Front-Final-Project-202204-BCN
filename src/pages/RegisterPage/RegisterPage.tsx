import Header from "../../components/Header/Header";
import RegisterForm from "../../components/RegisterForm/RegisterForm";

const RegisterPage = (): JSX.Element => {
  return (
    <>
      <Header pageName={"Register"} adminUser={false} />
      <RegisterForm />
    </>
  );
};

export default RegisterPage;
