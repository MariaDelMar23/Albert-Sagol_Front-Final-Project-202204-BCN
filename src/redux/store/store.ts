import { configureStore } from "@reduxjs/toolkit";
import petsReducer from "../features/petsSlice";
import userReducer from "../features/userSlice";

const store = configureStore({
  reducer: {
    user: userReducer,
    pets: petsReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
