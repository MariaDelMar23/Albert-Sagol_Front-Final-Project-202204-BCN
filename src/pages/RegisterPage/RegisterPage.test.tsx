import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../../redux/store/store";
import RegisterPage from "./RegisterPage";

describe("Given the RegisterPage page", () => {
  describe("When it's called", () => {
    test("Then it should render a heading with the text 'Register'", () => {
      render(
        <Provider store={store}>
          <RegisterPage />
        </Provider>
      );

      expect(
        screen.getByRole("heading", { level: 1, name: "Register" })
      ).toBeInTheDocument();
    });

    test("Then it should render a label with 'Name'", () => {
      render(
        <Provider store={store}>
          <RegisterPage />
        </Provider>
      );

      expect(screen.getByLabelText("Name")).toBeInTheDocument();
    });
  });
});
