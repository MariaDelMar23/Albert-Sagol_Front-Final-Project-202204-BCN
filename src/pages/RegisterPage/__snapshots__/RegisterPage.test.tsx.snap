// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the RegisterPage page When it's called and there are a token on the local Storage Then it should match with the snapshot 1`] = `
Array [
  <header
    className="sc-bczRLJ QbMXw"
  >
    <div
      className="container--header-logo-title"
    >
      <img
        alt="Pet's House logo"
        height={74}
        src="./images/logo.svg"
        width={152}
      />
    </div>
    <h1>
      Register
    </h1>
    <div
      className="container--header-buttons"
    />
  </header>,
  <form
    autoComplete="off"
    className="sc-gsnTZi brwyZO register-form"
    noValidate={true}
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="name"
      >
        Name
        <input
          id="name"
          onChange={[Function]}
          placeholder="Name"
          type="text"
          value=""
        />
      </label>
      <label
        htmlFor="username"
      >
        Username
        <input
          id="username"
          onChange={[Function]}
          placeholder="Username"
          type="text"
          value=""
        />
      </label>
      <label
        htmlFor="eMail"
      >
        Email
        <input
          id="eMail"
          onChange={[Function]}
          placeholder="eMail"
          type="text"
          value=""
        />
      </label>
      <label
        htmlFor="password"
      >
        Password
        <input
          id="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </label>
      <button
        disabled={true}
        type="submit"
      >
        Register
      </button>
    </div>
    <div>
      <p>
        You have an account
      </p>
      <a
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </div>
  </form>,
]
`;
